components/collector/tests/source_collectors/jenkins/test_failed_jobs.py

```diff
@@ -137,3 +143,40 @@ async def test_no_builds(self):
         }
         response = await self.collect(get_request_json_return_value=jenkins_json)
         self.assert_measurement(response, entities=[])
+
+    async def test_ignore_jobs_by_grace_period_within_period(self):
+        """Test that jobs can be ignored by grace period."""
+        self.set_source_parameter("grace_days", "5")
+        builds = [
+            {"result": "FAILURE", "timestamp": self.timestamp()},
+            {"result": "SUCCESS", "timestamp": self.timestamp(days_ago=4)},
```

Comment:
Shouldn't this also be 6 days ago, in order to test that the short-circuit works?

Reply:
I'll add a few more variants.